packages/medusa/src/services/product-collection.js

```diff
@@ -107,15 +107,47 @@ class ProductCollectionService extends BaseService {
         this.productCollectionRepository_
       )
 
+      const { images, ...rest } = collection
+      if (!rest.thumbnail && images && images.length) {
+        rest.thumbnail = images[0]
+      }
+
       try {
         const productCollection = await collectionRepo.create(collection)
+
+        if (images) {
+          productCollection.images = await this.upsertImages_(images)
```

Comment:
You are re-implementing a method that already exists in the ImageRepository. 

Instead, you could just do like this:

```ts
const imageRepo = manager.getCustomRepository(this.imageRepository_)

// ...

if (images?.length) {
  productCollection.images = await imageRepo.upsertImages_(images)
}
```